librosa/core/pitch.py

```diff
@@ -335,3 +338,281 @@ def piptrack(y=None, sr=22050, S=None, n_fft=2048, hop_length=None,
+     y : np.ndarray [shape=(n,)]
+         audio time series.
+
+     sr : number > 0 [scalar]
+         sampling rate of `y` in Hertz.
+
+     win_length : int [scalar]
+         length of the window in samples.
+         By default, win_length=1024 corresponds to a time scale of 46 ms at
+         a sampling rate of 22050 Hz.
+
+     hop_length : None or int > 0 [scalar]
+         number of audio samples between adjacent pYIN predictions.
+         If `None`, defaults to `frame_length // 4`.
+
+     fmin: number > 0 [scalar]
+         minimum frequency in Hertz.
+
+     fmax: number > 0 [scalar]
+         maximum frequency in Hertz.
+
+     peak_threshold: number > 0 [scalar]
+         absolute threshold for peak estimation.
+
+     Returns
+     -------
+     f0: np.ndarray [shape=(n_frames,)]
+         time series of fundamental frequencies in Hertz.
+
+     Examples
+     --------
+     Computing a fundamental frequency curve from an audio input:
+     >>> y = librosa.chirp(440, 880, duration=7.0)
+     >>> yin(y)
+     array([441.15076669, 441.6640896 , 442.16219657, ...,
+        871.59470743, 872.63699131, 873.66696896])
+     '''
+    # Set the default hop if it is not already specified.
+    if hop_length is None:
+        hop_length = int(win_length // 4)
+
+    # Check that audio is valid.
+    util.valid_audio(y, mono=True)
+
+    # Autocorrelation.
+    y_frames = util.frame(y, frame_length=2*win_length, hop_length=hop_length)
+    n_frames = y_frames.shape[-1]
+    a = np.fft.fft(y_frames, 2*win_length, axis=0)
+    b = np.fft.fft(y_frames[win_length::-1, :], 2*win_length, axis=0)
+    acf_frames = np.fft.ifft(a*b, 2*win_length, axis=0).real[win_length:]
+
+    # Energy terms.
+    window = scipy.signal.windows.boxcar(win_length + 1)
+    energy_frames = scipy.signal.fftconvolve(
+        y_frames ** 2, window[:, None], mode="valid", axes=0)
+
+    # Difference function.
+    yin_frames = energy_frames[0, :] + energy_frames - 2*acf_frames
+
+    # Cumulative mean normalized difference function.
+    min_period = np.maximum(int(np.floor(sr / fmax)), 2)
+    max_period = np.minimum(int(np.ceil(sr / fmin)), win_length - 1)
+    yin_numerator = yin_frames[min_period-1:max_period+1, :]
+    tau_range = np.arange(1, max_period+1)[:, None]
+    cumulative_mean = np.cumsum(yin_frames[1:max_period+1, :], axis=0) / tau_range
+    yin_denominator = cumulative_mean[min_period-2:max_period, :]
+    yin_frames = yin_numerator / (yin_denominator + util.tiny(yin_denominator))
+
+    # Parabolic interpolation.
+    parabola_a = (yin_frames[:-2, :] + yin_frames[2:, :] - 2*yin_frames[1:-1, :]) / 2
+    parabola_b = (yin_frames[2:, :] - yin_frames[:-2, :]) / 2
+    parabolic_shifts = -parabola_b / (2*parabola_a + util.tiny(parabola_a))
+    parabolic_shifts[np.abs(parabolic_shifts) > 1] = 0
+
+    # Find local minima.
+    is_trough = np.logical_and(
+        yin_frames[1:-1, :] < yin_frames[:-2, :],
+        yin_frames[1:-1, :] < yin_frames[2:, :])
```

Comment:
this seems okay, though it would miss edge cases by requiring strict minima.  Would it make sense to use our existing `localmax` function for this instead?

Reply:
I missed the `localmax` function. Happy to use that instead.